packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatTypeAssertions.ts

```diff
@@ -0,0 +1,97 @@
+import { CallWithChatAdapterManagement } from './CallWithChatAdapter';
+/* @conditional-compile-remove(file-sharing) */
+import { FileUploadAdapter } from '../../ChatComposite';
+
+/// IMPORTANT
+///
+/// Each interface defined in here is the pick&omit version of the flattened interface we export.
+/// We only export flattened apis for improved developer experience. However we need to ensure
+/// that the signatures that callwithchat uses matches the underlying signatures in the call
+/// and chat interfaces. Thus here we define interfaces pick and omit from the call and chat
+/// apis and assert that the flattened apis extend the signature.
+///
+/// This file will need kept up to date when changes to the flattened interfaces are made.
+
+/// CallWithChatAdapterManagement
+
+interface CallWithChatAdapterManagementInternal
+  extends Omit<CallAdapterCallManagement, 'removeParticipant'>,
+    CallAdapterDeviceManagement,
+    Omit<ChatAdapterThreadManagement, 'removeParticipant' | 'setTopic'>,
+    /* @conditional-compile-remove(file-sharing) */
+    FileUploadAdapter {}
+
+const CallWithChatAdapterManagementTypeAssertion = (
+  value: CallWithChatAdapterManagement
+): CallWithChatAdapterManagementInternal => value;
+
+const CallWithChatAdapterManagementRequiredTypeAssertion = (
+  value: Required<CallWithChatAdapterManagement>
+): Required<CallWithChatAdapterManagementInternal> => value;
+
+CallWithChatAdapterManagementTypeAssertion;
+CallWithChatAdapterManagementRequiredTypeAssertion;
+
+/// CallWithChatControlOptions
+
+interface CallWithChatControlOptionsInternal
+  extends Omit<
+    CallControlOptions,
+    | 'endCallButton'
+    | 'devicesButton'
+    | /* @conditional-compile-remove(control-bar-button-injection) */ 'onFetchCustomButtonProps'
+    | 'participantsButton'
+  > {}
+
+const CallWithChatControlOptionsTypeAssertion = (
+  value: CallWithChatControlOptions
+): CallWithChatControlOptionsInternal => value;
+
+const CallWithChatControlOptionsRequiredTypeAssertion = (
+  value: Required<CallWithChatControlOptions>
+): Required<CallWithChatControlOptionsInternal> => value;
+
+CallWithChatControlOptionsTypeAssertion;
+CallWithChatControlOptionsRequiredTypeAssertion;
+
+/// CallWithChatAdapterUiState
+
+interface CallWithChatAdapterUiStateInternal extends CallAdapterUiState, Omit<ChatAdapterUiState, 'error'> {}
+
+const CallWithChatAdapterUiStateTypeAssertion = (
+  value: CallWithChatAdapterUiState
+): CallWithChatAdapterUiStateInternal => value;
+
+const CallWithChatAdapterUiStateRequiredTypeAssertion = (
+  value: Required<CallWithChatAdapterUiState>
+): Required<CallWithChatAdapterUiStateInternal> => value;
+
+CallWithChatAdapterUiStateTypeAssertion;
+CallWithChatAdapterUiStateRequiredTypeAssertion;
+
+/// CallWithChatClientState
+
+interface CallWithChatClientStateInternal
+  extends Omit<CallAdapterClientState, 'displayName' | 'endedCall' | 'latestErrors' | 'userId'> {}
```

Comment:
Omit style assertions are easier to maintain -- whenever we add a new field to the base type, it will break this assertion, and we need to explicitly Omit it if we intend to not add to CallWithChat.